cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/SchemaContainer.tsx

```diff
@@ -171,6 +188,105 @@ class SchemaContainer extends React.Component<ISchemaContainerProps, ISchemaCont
     return validationError;
   }
 
+  private componentID = `sql-fields-${uuidV4()}`;
+
+  private io = new IntersectionObserver(
+    (entries) => {
+      let lastVisibleElement = this.state.fieldWindowSize;
+      for (const entry of entries) {
+        let id = entry.target.getAttribute('id');
+        id = id.split('-').pop();
+        const idInt = parseInt(id, 10);
+        if (entry.isIntersecting) {
+          lastVisibleElement =
+            idInt + 50 > this.state.fieldWindowSize ? idInt + DEFAULT_FIELD_WINDOW_SIZE : idInt;
```

Comment:
I assume the 50 here is coming from DEFAULT_FIELD_WINDOW_SIZE? If yes, can we use the variable here instead?